pkg/kubelet/cm/devicemanager/endpoint.go

```diff
@@ -35,7 +35,7 @@ import (
 type endpoint interface {
 	run()
 	stop()
-	allocate(devs []string) (*pluginapi.AllocateResponse, error)
+	allocate(devs []string, podAnnotations map[string]string) (*pluginapi.AllocateResponse, error)
```

Comment:
I'm wondering if It might be worth just passing a pluginapi.AllocateRequest, what do you think?

Reply:
> Aren't you missing the part where you actually pass the pod annotations to the manager?

No, I'm not missing it I think. It's in manager.go:

```
-		resp, err := e.allocate(devs)
+		resp, err := e.allocate(devs, pod.Annotations)
```

Reply:
> I'm wondering if It might be worth just passing a pluginapi.AllocateRequest, what do you think?

I thought about it and decided to use ContainerAllocateRequest as there is a chance that at some point this call might get ContainerAllocateRequests from different pods. If it's not designed this way I'd be happy to update this PR.

Reply:
> No, I'm not missing it I think. It's in manager.go:

Yep missed that, thanks for highlighting it :)

> I thought about it and decided to use ContainerAllocateRequest as there is a chance that at some point this call might get ContainerAllocateRequests from different pods. If it's not designed this way I'd be happy to update this PR.

So there are multiple things:
- in the flow of the pod's lifecycle the n-Allocate calls are called as part of the pod admission process, which goes through the pods in sequential order
- we expect to merge the n-Allocate calls to a single call (`PodAdmission`?), though that's not yet decided

Reply:
Yes, I saw that currently allocate call is called sequentially for every init container and container of the pod. I just thought that AllocateRequest can potentially have multiple ContainerAllocateRequests, so it would be safer to pass pod annotations through ContainerAllocateRequest.

So, should I follow your suggestion and put pod allocations into pluginapi.AllocateRequest ?

Reply:
That's ok, we'll probably change that call anyways so I think it's ok in the current form